gmso/core/improper.py

```diff
@@ -0,0 +1,55 @@
+import warnings
+
+from gmso.core.connection import Connection
+from gmso.core.improper_type import ImproperType
+from gmso.exceptions import GMSOError
+
+
+class Improper(Connection):
+    """A 4-partner connection between sites.
+
+    This is a subclass of the gmso.Connection superclass.
+    This class has strictly 4 members in its connection_members.
+    The connection_type in this class corresponds to gmso.ImproperType
```

Comment:
Might be worth defining what make a dihedral improper, for those with less domain knowledge. Maybe some ASCII art?